treeherder/etl/job_loader.py

```diff
@@ -239,6 +239,22 @@ def _get_log_references(self, job):
                 "url": logref["url"],
                 "parse_status": "parsed" if "steps" in logref else "pending"
             })
+        log_references.extend(self._get_errorsummary_log_references(job))
+        return log_references
+
+    def _get_errorsummary_log_references(self, job):
+        log_references = []
+        if "jobInfo" in job:
+            ji = job["jobInfo"]
+            if "links" in ji:
+                links = ji["links"]
+                for link in links:
+                    if ("linkText" in link and
+                        link["linkText"].endswith("_errorsummary.log")
+                        and "url" in link):
+                        log_references.append({"name": "errorsummary_json",
+                                               "url": link["url"],
+                                               "parse_status": "pending"})
```

Comment:
This can be written as:
```python
log_references = []
links = job.get("jobinfo", {}).get("links", [])
for link in links:
    url = link.get("url")
    linktext = link.get("linkText", "")
    if url and linktext.endswith("_errorsummary.log"):
        log_references.append({"name": "errorsummary_json",
                               "url": url,
                               "parse_status": "pending"})
```
(Or if you're not keen on the `linktext` default of the empty string, then `if url and linktext and linktext.endswith("_errorsummary.log")` is still cleaner)